### PR TITLE
rewritten .then/.catch statements to async await functions

### PR DESCRIPTION
**Pull request by JonHaxha**

### CODE REVIEW

File: src/lib/api/api.ts [hunk 1]
```diff
@@ -1215,36 +1162,41 @@ export const checkIfItemIsSharedForRename = ({ type, uuid, metaData }: { type: s
...
 
-export const fileExists = ({ name, parent }: { name: string; parent: string }): Promise<{ exists: boolean; existsUUID: string }> => {
+export const fileExists = async ({
+	name,
+	parent
+}: {
+	name: string
+	parent: string
+}): Promise<{ exists: boolean; existsUUID: string }> => {
 	return new Promise((resolve, reject) => {
```
Comment (Dwynr) [inline src/lib/api/api.ts:1200]: missing refactor to async/await

File: src/lib/api/api.ts [hunk 2]
```diff
@@ -1466,194 +1428,179 @@ export const createFolder = ({ uuid, name, parent, emitEvents = true }: { uuid:
...
+	}
 
-								return resolve(true)
-							})
-							.catch(reject)
-					})
-					.catch(reject)
-			})
-			.catch(reject)
+	checkIfItemParentIsShared({
```
Comment (Dwynr) [inline src/lib/api/api.ts:1522]: "await" missing

File: src/lib/api/api.ts [hunk 3]
```diff
@@ -1466,194 +1428,179 @@ export const createFolder = ({ uuid, name, parent, emitEvents = true }: { uuid:
...
```
Comment (Dwynr) [inline src/lib/api/api.ts:1586]: "await" missing

File: src/lib/api/api.ts [hunk 4]
```diff
@@ -1466,194 +1428,179 @@ export const createFolder = ({ uuid, name, parent, emitEvents = true }: { uuid:
...
-						})
-							.then(() => {
-								if (emitEvents) {
-									eventListener.emit("itemMoved", {
-										item: folder,
-										from: folder.parent,
-										to: parent
-									})
-								}
+		return response.message
```
Comment (Dwynr) [inline src/lib/api/api.ts:1583]: throw error

File: src/lib/api/api.ts [hunk 5]
```diff
@@ -1733,64 +1680,56 @@ export const renameFile = ({ file, name }: { file: ItemProps; name: string }): P
...
-										return resolve(true)
-									})
-									.catch(reject)
-							})
-							.catch(reject)
-					})
-					.catch(reject)
-			})
-			.catch(reject)
+	checkIfItemIsSharedForRename({
```
Comment (Dwynr) [inline src/lib/api/api.ts:1719]: "await" missing

**Review by Dwynr — CHANGES_REQUESTED** [review]

File: src/lib/api/api.ts (hunk 1 above)
Comment (JonHaxha) [reply src/lib/api/api.ts:1200]: refactored

File: src/lib/api/api.ts [hunk 6]
```diff
@@ -2203,118 +2113,91 @@ export const shareItemsToUser = async ({
...
+		method: "POST",
+		endpoint: "/v1/user/publicKey/get",
+		data: {
+			email
+		}
+	})
+	if (!response.status) {
+		throw new Error(response.message)
+	}
+	return response.data
```
Comment (Dwynr) [inline src/lib/api/api.ts:2135]: response.data.publicKey

**Pull request closed without merging**